R/RLearner_classif_fdaFDboost.R

```diff
@@ -0,0 +1,116 @@
+#' @export
+makeRLearner.classif.fdaFDboost = function() {
+  makeRLearnerClassif(
+    cl = "classif.fdaFDboost",
+    package = c("FDboost", "mboost"),
+    par.set = makeParamSet(
+      makeDiscreteLearnerParam(id = "family", default = "Binomial", values = c("AdaExp", "Binomial", "AUC", "custom.family")),
+      makeIntegerLearnerParam(id = "mstop", default = 100L, lower = 1L),
+      makeNumericLearnerParam(id = "nu", default = 0.1, lower = 0, upper = 1),  # the learning rate
+      makeUntypedLearnerParam(id = "custom.family.definition", requires = quote(family == "custom.family")),  # list of parameters for the custom family
+      # makeDiscreteLearnerParam(id = "risk", values = c("inbag", "oobag", "none")), we don't need this in FDboost
+      makeDiscreteLearnerParam(id = "Binomial.link", default = "logit",
+        values = c("logit", "probit"), requires = quote(family == "Binomial")),
+      makeNumericLearnerParam(id = "df", default = 4, lower = 0.5),  # effective degrees of freedom, depend on the regularization parameter of the penality matrix and number of splines, must be the same for all base learners(covariates), the maximum value is the rank of the design matrix
+      # makeDiscreteLearnerParam(id = "baselearner", values = c("bbs", "bols")),  # we don't use "btree" in FDboost
+      makeIntegerLearnerParam(id = "knots", default = 10L, lower = 1L),  # determine the number of knots of splines, does not matter once there is sufficient number of knots, 30,40, 50 for example
+      makeIntegerLearnerParam(id = "degree", default = 3L, lower = 1L),  # degree of the b-spline
+      makeIntegerLearnerParam(id = "differences", default = 1L, lower = 1L),  # degree of the penalty
+      makeLogicalLearnerParam(id = "bsignal.check.ident", default = FALSE, tunable = FALSE)  # identifiability check by testing matrix degeneracy
+    ),
+    par.vals = list(family = "Binomial"),
+    properties = c("functionals", "numerics", "twoclass"),
+    name = "Functional linear array classification boosting",
+    short.name = "FDboost",
+    note = "Only allow one base learner for functional covariate and one base learner for scalar covariate, the parameters for these base learners are the same. Also we currently do not support interaction between scalar covariates"
```

Comment:
Only allows one base learner for each functional covariate and one base learner for each scalar covariate, the parameters for these base learners are the same. Also we currently do not support interaction between scalar covariates.

Reply:
Update note for classif and regr